### PR TITLE
Keeps page

### PR DESCRIPTION
**Pull request by IgorOsiyuk**
create current keeps and list of keeps
also created request for get keeps from server
created actions and reducers for keeps request

image

### CODE REVIEW

**Comment by melonholy** [comment]
image
Weird placeholder names

**Comment by melonholy** [comment]
Add. behavior of closing modal after click outside the modal

**Comment by melonholy** [comment]
Cant check functionality 
image
You should change backend url for deployed version

File: google_keeps/src/components/keeps/Keep.jsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
+import styled, { keyframes } from "styled-components";
+import { Delete } from "@styled-icons/feather/Delete";
+import { Send } from "@styled-icons/fluentui-system-regular/Send";
+import { StyledIconBase } from "@styled-icons/styled-icon";
+
+const ControlPanel = styled.div`
+  height: 40px;
+  width: 100%;
+  position: absolute;
```
Comment (melonholy) [inline google_keeps/src/components/keeps/Keep.jsx:9]: You don't need position absolute for this

File: google_keeps/src/components/keeps/Keep.jsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    transform: scale3d(1, 1, 1);
+  }
+`;
+
+const KeepWrap = styled.div`
+  width: 25%;
+  padding: 16px;
+  animation-name:${keepShow};
+  animation-duration:1s;
+  animation-fill-mode:forwards;
```
Comment (melonholy) [inline google_keeps/src/components/keeps/Keep.jsx:72]: spaces

File: google_keeps/src/components/keeps/Keep.jsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+`;
+
+const KeepForm = styled.div`
+  width: 100%;
+  padding-bottom: 40px;
+  border-radius: 8px;
+  border: 1px solid transparent;
+  background-color: #fff;
```
Comment (melonholy) [inline google_keeps/src/components/keeps/Keep.jsx:80]: You have theme color for this

File: google_keeps/src/components/keeps/Keep.jsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+const TextInput = styled(KeepInput)`
+  font-size: 14px;
+  font-weight: 400;
+  min-height: 30px;
+  padding: 4px 16px 12px 16px;
+`;
+
+const DeleteIcon = styled(Delete)``;
+
+const UpdateIcon = styled(Send)``;
```
Comment (melonholy) [inline google_keeps/src/components/keeps/Keep.jsx:120]: Why do you need empty components?

File: google_keeps/src/components/keeps/Keep.jsx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+  padding: 5px;
+  ${StyledIconBase} {
+    height: 20px;
+    width: 20px;
+  }
+  &:hover {
+    background-color: #dedede;
```
Comment (melonholy) [inline google_keeps/src/components/keeps/Keep.jsx:141]: Theme color variable

File: google_keeps/src/components/keeps/Keep.jsx [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+  }
+  & + & {
```
Comment (melonholy) [inline google_keeps/src/components/keeps/Keep.jsx:143]: ???????

File: google_keeps/src/components/keeps/KeepsList.jsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  margin: 0 -16px;
+`;
+
+const KeepsList = ({ list }) => {
+  return (
+    <KeepListContainer>
+      {list.length !== 0 ? (
```
Comment (melonholy) [inline google_keeps/src/components/keeps/KeepsList.jsx:14]: Just !list.length

File: google_keeps/src/redux/action/types.jsx [hunk 8]
```diff
@@ -1,3 +1,5 @@
 export const AUTHORIZATION_USER = "AUTHORIZATION_USER";
 export const AUTHORIZATION_ERROR = "AUTHORIZATION_ERROR";
 export const AUTHORIZATION_ERROR_CLEANER = "AUTHORIZATION_ERROR_CLEANER";
+
+export const GET_USER_KEEPS = "GET_USER_KEEPS";
```
Comment (melonholy) [inline google_keeps/src/redux/action/types.jsx:5]: New line

File: google_keeps/src/redux/action/user.jsx [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    type: actionType.GET_USER_KEEPS,
+    payload: data,
+  };
+}
+
+export function GetUserKeepsRequest() {
+  return (dispatch) => {
+    return UserService.getUserKeepsList().then((res) =>
+      dispatch(GetUserKeeps(res.data))
+    );
```
Comment (melonholy) [inline google_keeps/src/redux/action/user.jsx:15]: no catch?

File: google_keeps/src/services/auth-header.jsx [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import LocalStorageService from './localStorage.service'
+import LocalStorageService from "./localStorage.service";
```
Comment (melonholy) [inline google_keeps/src/services/auth-header.jsx:1]: Services again)

**Review by melonholy — CHANGES_REQUESTED** [review]

File: google_keeps/src/components/keeps/Keep.jsx (hunk 1 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/keeps/Keep.jsx]: - [+]

File: google_keeps/src/components/keeps/Keep.jsx (hunk 4 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/keeps/Keep.jsx:127]: removed it. I write it for future maybe I change something in this case
- [+]

File: google_keeps/src/components/keeps/Keep.jsx (hunk 5 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/keeps/Keep.jsx:150]: - [+]

File: google_keeps/src/components/keeps/Keep.jsx (hunk 6 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/keeps/Keep.jsx:152]: what's the question? ) 
I used "justify-content: flex-end" and each of the next button after first has margin-left 20px

File: google_keeps/src/components/keeps/KeepsList.jsx (hunk 7 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/keeps/KeepsList.jsx:14]: - [+]

File: google_keeps/src/redux/action/types.jsx (hunk 8 above)
Comment (IgorOsiyuk) [reply google_keeps/src/redux/action/types.jsx]: - [+]

File: google_keeps/src/services/auth-header.jsx (hunk 10 above)
Comment (IgorOsiyuk) [reply google_keeps/src/services/auth-header.jsx:1]: - [+]

File: google_keeps/src/components/keeps/Keep.jsx (hunk 1 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/keeps/Keep.jsx]: - [+]

File: google_keeps/src/components/keeps/Keep.jsx (hunk 2 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/keeps/Keep.jsx]: - [+]

File: google_keeps/src/components/keeps/Keep.jsx (hunk 3 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/keeps/Keep.jsx]: - [+]

File: google_keeps/src/redux/action/user.jsx (hunk 9 above)
Comment (IgorOsiyuk) [reply google_keeps/src/redux/action/user.jsx:15]: I decided that add error functional when I'm going to do notification components

File: google_keeps/src/components/keeps/KeepsList.jsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+  return (
+    <KeepListContainer>
+      {!!list.length ? (
+        list.map((item) => {
+          return <Keep data={item} key={item._id} />;
+        })
+      ) : (
+        <h2>у вас нет заметок</h2>
+      )}
+    </KeepListContainer>
```
Comment (melonholy) [inline google_keeps/src/components/keeps/KeepsList.jsx:21]: Better write smth like this 
```
  if (!list.length)
    return (
      <KeepListContainer>
        <h2>у вас нет заметок</h2>
      </KeepListContainer>
    )

  return (
    <KeepListContainer>
      {list.map(item =>
        <Keep data={item} key={item._id} />
      )}
    </KeepListContainer>
  )
```

File: google_keeps/src/components/keeps/Keep.jsx [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+  overflow: hidden;
+  min-height: 120px;
+  transition-duration: 0.218s;
+  transition-property: background, border, opacity, box-shadow, transform;
+  transition-timing-function: ease-in;
+  position: relative;
+  &:hover {
+    box-shadow: 0 1px 2px 0 rgba(60, 64, 67, 0.3),
+      0 1px 3px 1px rgba(60, 64, 67, 0.15);
+    ${ControlPanel} {
```
Comment (melonholy) [inline google_keeps/src/components/keeps/Keep.jsx:50]: Put new line between separate elements. Also check for all files

File: google_keeps/src/components/keeps/KeepsList.jsx [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import styled from "styled-components";
+import Keep from "./Keep";
+
+const KeepListContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  margin: 0 -16px;
```
Comment (melonholy) [inline google_keeps/src/components/keeps/KeepsList.jsx:8]: Why do you need negative margins?

**Review by melonholy — CHANGES_REQUESTED** [review]

File: google_keeps/src/components/keeps/KeepsList.jsx (hunk 11 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/keeps/KeepsList.jsx:21]: - [+]

File: google_keeps/src/components/keeps/Keep.jsx (hunk 12 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/keeps/Keep.jsx:50]: - [+]

File: google_keeps/src/components/keeps/KeepsList.jsx (hunk 13 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/keeps/KeepsList.jsx:8]: - [+]

**Review by melonholy — APPROVED** [review]

**Pull request merged**